sdk/keyvault/keyvault/src/az_keyvault_client.c

```diff
@@ -176,20 +180,30 @@ AZ_NODISCARD az_result _az_keyvault_keys_key_create_build_json_body(
   return AZ_OK;
 }
 
+AZ_NODISCARD static az_result _az_reset_url_to_initial_state(az_keyvault_keys_client * client) {
+  if (client->_internal.initial_url_length != az_span_length(client->_internal.uri)) {
+    // Can't use slice here because we would lost original capacity
+    client->_internal.uri = az_span_init(
+        az_span_ptr(client->_internal.uri),
+        client->_internal.initial_url_length,
+        az_span_capacity(client->_internal.uri));
+  }
+  return AZ_OK;
+}
+
 AZ_NODISCARD az_result az_keyvault_keys_key_create(
     az_keyvault_keys_client * client,
     az_span key_name,
     json_web_key_type json_web_key_type,
     az_keyvault_create_key_options * options,
     az_http_response * response) {
 
-  // Request buffer
-  uint8_t url_buffer[1024 * 2];
+  // Headers buffer
   uint8_t headers_buffer[4 * sizeof(az_pair)];
-  az_span request_url_span = AZ_SPAN_FROM_BUFFER(url_buffer);
   az_span request_headers_span = AZ_SPAN_FROM_BUFFER(headers_buffer);
 
-  /* ******** build url for request  ******/
+  // check if url needs to be reset to initial state
+  AZ_RETURN_IF_FAILED(_az_reset_url_to_initial_state(client));
```

Comment:
I don't understand this. When an operation (like key_create) is started, a new url_buffer is allocated on the stack and the client's url is copied to it - period. There is no "resetting" of anything because it is already brand new.